File: ReinforcementLearning/ver0.2/Airport.py
class Airport:
    # deadheadCost = dict('name2', 'deadhead') # dictionary 자료형, name/deadhead로 인덱싱

    def __init__(self, id, name, hotelCost):
        self.id = id
        self._name = name  # 공항 이름
        self._hotelCost = hotelCost

    @property
    def name(self):
        return self._name

    @name.setter
    def name(self, value):
        self._name = value

    @property
    def hotelCost(self):
        return self._hotelCost

    @hotelCost.setter
    def hotelCost(self, value):
        self._hotelCost = value

    # @property
    # def deadheadCost(self):
    #   return self.deadheadCost

    # @deadheadCost.setter
    # def deadheadCost(self, value):
    #   self.deadheadCost = value

    # 인스턴스 이름 설정 (java에서 toString())

    def __str__(self):
        return f"Airport - {self.name}"

    def find_airport(self, airports, name):
        airport = next(
            (airport for airport in airports if airport.name == name), None)
        if airport is None:
            raise ValueError("Airport not found")
        return airport

File: ManagerProgram/src/main/java/org/dongguk/domain/Airport.java
package org.dongguk.domain;

import lombok.*;
import org.dongguk.AbstractPersistable;

import java.util.List;
import java.util.Map;

@Getter
@Setter
@AllArgsConstructor
public class Airport extends AbstractPersistable {
    // 공항 이름
    private String name;
    // 공항에 따른 호텔비용
    private int hotelCost;
    // 공항에 따른 Deadhead Cost 맵 ex) deadheadCost.get("ATL") -> 200
    private Map<String, Integer> deadheadCost;

    @Override
    public String toString() {
        return "Airport - " + name;
    }

    public static Airport of(List<Airport> airports, String name) {
        return airports.stream()
                .filter(airport -> airport.getName().equals(name))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Airport not found"));
    }

    @Builder
    public Airport(long id, String name, int hotelCost, Map<String, Integer> deadheadCost) {
        super(id);
        this.name = name;
        this.hotelCost = hotelCost;
        this.deadheadCost = deadheadCost;
    }

    public void putDeadhead(String name, Integer deadhead) {
        deadheadCost.put(name, deadhead);
    }
}
